internal/kibana/reconciler.go

```diff
@@ -421,12 +430,23 @@ func getImage() string {
 	return utils.LookupEnvWithDefault("KIBANA_IMAGE", kibanaDefaultImage)
 }
 
-func getProxyImage() string {
-	return utils.LookupEnvWithDefault("PROXY_IMAGE", kibanaProxyDefaultImage)
+func getProxyImage(ctx context.Context, c client.Client) (string, error) {
+	// Get the existing imagestream
+	key := client.ObjectKey{Name: oauthProxyImageName, Namespace: oauthProxyNamespace}
+	is, err := img.Get(ctx, c, key)
+	if err != nil {
+		return "", err
+	}
+	if len(is.Status.Tags) != 0 {
+		if len(is.Status.Tags[0].Items) != 0 {
+			return is.Status.Tags[0].Items[0].DockerImageReference, nil
+		}
+	}
+	return "", kverrors.New("Failed to pull oauth-proxy image spec")
```

Comment:
Idiomatic Go usually avoids nesting error conditions if possible and does the "happy path" on the unindented level which makes the code easier to read, like this:

```suggestion
	if len(is.Status.Tags) == 0 {
		return "", errors.New("proxy ImageStream has no tags")
	}

	if len(is.Status.Tags[0].Items) == 0 {
		return "", errors.New("proxy ImageStream tag has no images")
	}

	return is.Status.Tags[0].Items[0].DockerImageReference, nil
```

This also makes it easier to provide detailed error messages for each possible failure.